### ISSUE TITLE
google-storage-wagon shades an old version of jackson, breaking git-commit-id-plugin

### ISSUE THREAD

**Issue by pcarrier** [gkatzioura/CloudStorageMaven#15]
An old version of Jackson is in the dependency tree:
```
[INFO] --- maven-dependency-plugin:2.8:tree (default-cli) @ google-storage-wagon ---
[INFO] com.gkatzioura.maven.cloud:google-storage-wagon:jar:1.4
[INFO] +- org.slf4j:jcl-over-slf4j:jar:1.7.25:compile
[INFO] |  \- org.slf4j:slf4j-api:jar:1.7.25:compile
[INFO] +- org.apache.maven.wagon:wagon-provider-api:jar:3.0.0:provided
[INFO] |  \- org.codehaus.plexus:plexus-utils:jar:3.0.24:provided
[INFO] +- com.google.cloud:google-cloud-storage:jar:1.20.0:compile
[INFO] |  +- com.google.cloud:google-cloud-core:jar:1.20.0:compile
[INFO] |  |  +- com.google.guava:guava:jar:20.0:compile
[INFO] |  |  +- joda-time:joda-time:jar:2.9.2:compile
[INFO] |  |  +- com.google.http-client:google-http-client:jar:1.23.0:compile
[INFO] |  |  |  \- org.apache.httpcomponents:httpclient:jar:4.0.1:compile
[INFO] |  |  |     +- org.apache.httpcomponents:httpcore:jar:4.0.1:compile
[INFO] |  |  |     +- commons-logging:commons-logging:jar:1.1.1:compile
[INFO] |  |  |     \- commons-codec:commons-codec:jar:1.3:compile
[INFO] |  |  +- com.google.code.findbugs:jsr305:jar:3.0.1:compile
[INFO] |  |  +- com.google.api:api-common:jar:1.4.0:compile
[INFO] |  |  +- com.google.api:gax:jar:1.19.0:compile
[INFO] |  |  |  \- org.threeten:threetenbp:jar:1.3.3:compile
[INFO] |  |  +- com.google.protobuf:protobuf-java-util:jar:3.5.1:compile
[INFO] |  |  |  +- com.google.protobuf:protobuf-java:jar:3.5.1:compile
[INFO] |  |  |  \- com.google.code.gson:gson:jar:2.7:compile
[INFO] |  |  +- com.google.api.grpc:proto-google-common-protos:jar:1.2.0:compile
[INFO] |  |  \- com.google.api.grpc:proto-google-iam-v1:jar:0.3.0:compile
[INFO] |  +- com.google.cloud:google-cloud-core-http:jar:1.20.0:compile
[INFO] |  |  +- com.google.auth:google-auth-library-credentials:jar:0.9.0:compile
[INFO] |  |  +- com.google.auth:google-auth-library-oauth2-http:jar:0.9.0:compile
[INFO] |  |  +- com.google.oauth-client:google-oauth-client:jar:1.23.0:compile
[INFO] |  |  +- com.google.api-client:google-api-client:jar:1.23.0:compile
[INFO] |  |  +- com.google.http-client:google-http-client-appengine:jar:1.23.0:compile
[INFO] |  |  +- com.google.http-client:google-http-client-jackson:jar:1.23.0:compile
[INFO] |  |  |  \- org.codehaus.jackson:jackson-core-asl:jar:1.9.11:compile
[INFO] |  |  +- com.google.http-client:google-http-client-jackson2:jar:1.23.0:compile
[INFO] |  |  |  \- com.fasterxml.jackson.core:jackson-core:jar:2.1.3:compile
[INFO] |  |  +- com.google.api:gax-httpjson:jar:0.36.0:compile
[INFO] |  |  +- io.opencensus:opencensus-api:jar:0.11.1:compile
[INFO] |  |  |  +- com.google.errorprone:error_prone_annotations:jar:2.2.0:compile
[INFO] |  |  |  \- io.grpc:grpc-context:jar:1.9.0:compile
[INFO] |  |  \- io.opencensus:opencensus-contrib-http-util:jar:0.11.1:compile
[INFO] |  \- com.google.apis:google-api-services-storage:jar:v1-rev114-1.23.0:compile
[INFO] +- com.gkatzioura.maven.cloud:cloud-storage-core:jar:1.4:compile
[INFO] +- commons-io:commons-io:jar:2.6:compile
[INFO] \- junit:junit:jar:4.12:test
[INFO]    \- org.hamcrest:hamcrest-core:jar:1.3:test
```

The version of Jackson is quite old, which breaks `git-commit-id-plugin`:

```
[INFO] Caused by: java.lang.NoSuchMethodError: com.fasterxml.jackson.core.JsonFactory.requiresPropertyOrdering()Z
[INFO]     at com.fasterxml.jackson.databind.ObjectMapper.<init> (ObjectMapper.java:571)
[INFO]     at com.fasterxml.jackson.databind.ObjectMapper.<init> (ObjectMapper.java:480)
[INFO]     at pl.project13.maven.git.GitCommitIdMojo.maybeGeneratePropertiesFile (GitCommitIdMojo.java:596)
```

jackson-core is indeed shaded but not relocated:
```
$ javap -s -cp ~/.m2/repository/com/gkatzioura/maven/cloud/google-storage-wagon/1.3/google-storage-wagon-1.3.jar com.fasterxml.jackson.core.JsonFactory
Compiled from "JsonFactory.java"
public class com.fasterxml.jackson.core.JsonFactory implements com.fasterxml.jackson.core.Versioned,java.io.Serializable {
[…]
```

**Comment by gkatzioura** [comment]
Ok this might be resolved either by using the latest jar google cloud (provided they use the latest one)
or exclude the dependency through you pom. I will try updating to the latest google jar provided nothing breaks.

**Comment by gkatzioura** [comment]
Updated dependency and shaded jackson, new release coming out.

**Comment by gkatzioura** [comment]
Released the fix on version 1.6
